src/main/java/eod/card/concrete/normal/GrandFeast.java

```diff
@@ -25,7 +29,9 @@ public GrandFeast() {
 
     @Override
     public void applyEffect(EffectExecutor executor) {
-        player.registerListener(this);
+        for(WarObject object:WarObject (player.getBoard()).which(OwnedBy(player)).which(Being(Character.class)).which(InParty(Party.RED)).get()) {
+            object.addStatus(Status.FURIOUS);
```

Comment:
In contexts where EffectExecutor is accessible, always consider using Effect methods.
They can enforce the effect flow through certain player, then player could have a chance to manipulate IO.

Reply:
fixed